filebeat/input/logv2/convert.go

```diff
@@ -0,0 +1,425 @@
+			return nil
+		}
+		if err := newCfg.SetChild(kind.fsName, -1, child); err != nil {
+			return fmt.Errorf("cannot set %q: %w", kind.fsName, err)
+		}
+	case ConfTypeConstant:
+		v, err := cfg.Bool(key, -1)
+		if err != nil {
+			logger.Warnf("cannot read %q as bool: %s, ignoring malformed config entry", key, err)
+			return nil
+		}
+		if v {
+			if err := newCfg.SetString(kind.fsName, -1, kind.fsVal); err != nil {
+				return fmt.Errorf("cannot set %q: %w", kind.fsName, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// handleMultilne converts the multiline configuration. There is at least one
+// field in the multiline configuration that cannot be directly copied, so
+// we have to call [translateField].
+func handleMultilne(logger *logp.Logger, cfg *config.C, parsers *[]any) error {
+	hasMultiline, err := cfg.Has("multiline", -1)
+	if err != nil {
+		return fmt.Errorf("cannot access 'multiline' field: %w", err)
+	}
+
+	if !hasMultiline {
+		return nil
+	}
+
+	newMultilineCfg := config.NewConfig()
+	multilineCfg, err := cfg.Child("multiline", -1)
+	if err != nil {
+		logger.Warnf("cannot read 'multiline' as map: %s, ignoring malformed config entry ", err)
+		return nil
+	}
+
+	count, err := multilineCfg.CountField("")
+	if err != nil {
+		logger.Warnf("cannot count elements from 'multiline': %s, ignoring malformed config entry", err)
+		return nil
+	}
+
+	// Ensure multiline is not an empty entry
+	if count > 0 {
+		for key, kind := range multilineConvTable {
+			has, err := multilineCfg.Has(key, -1)
+			if err != nil {
+				return fmt.Errorf("cannot read 'multiline.%s': %w", key, err)
+			}
+
+			if !has {
+				continue
+			}
+
+			if err := translateField(logger, multilineCfg, newMultilineCfg, key, kind); err != nil {
+				return err
+			}
+		}
+
+		*parsers = append(*parsers, map[string]any{
+			"multiline": multilineCfg,
+		})
+
+		return nil
+	}
+
+	return nil
+}
+
+// handleJSON copies the JSON configuration from the Log input into the
+// parsers array from Filestream
+func handleJSON(logger *logp.Logger, cfg *config.C, parsers *[]any) error {
+	hasJson, err := cfg.Has("json", -1)
+	if err != nil {
+		return fmt.Errorf("cannot read 'json': %w", err)
+	}
+
+	if !hasJson {
+		return nil
+	}
+
+	jsonCfg, err := cfg.Child("json", -1)
+	if err != nil {
+		logger.Warnf("cannot read 'json' as map: %s, ignoring malformed config entry ", err)
+		return nil
+	}
+
+	count, err := jsonCfg.CountField("")
+	if err == nil && count != 0 {
+		keysUnderRoot, err := jsonCfg.Bool("keys_under_root", -1)
+		if err != nil {
+			logger.Warnf(
+				"cannot read 'json.keys_under_root' as boolean: %s, ignoring malformed config entry ",
+				err,
+			)
+		}
+
+		if !keysUnderRoot {
+			if err := jsonCfg.SetString("target", -1, "json"); err != nil {
+				return fmt.Errorf("cannot set 'target' in the ndjson parser: %w", err)
+			}
+		}
+
+		*parsers = append(*parsers, map[string]any{"ndjson": jsonCfg})
+	}
```

Comment:
No warning if there is an error?

Also, same as above:


```suggestion
	count, err := jsonCfg.CountField("")
	if err != nil || count == 0 {
	return nil
	}
	
			keysUnderRoot, err := jsonCfg.Bool("keys_under_root", -1)
		if err != nil {
			logger.Warnf(
				"cannot read 'json.keys_under_root' as boolean: %s, ignoring malformed config entry ",
				err,
			)
		}

		if !keysUnderRoot {
			if err := jsonCfg.SetString("target", -1, "json"); err != nil {
				return fmt.Errorf("cannot set 'target' in the ndjson parser: %w", err)
			}
		}

		*parsers = append(*parsers, map[string]any{"ndjson": jsonCfg})

```